src/utils/generation_utils.py

```diff
@@ -452,3 +452,54 @@ def await_streamed_block(block: Block, context: AgentContext) -> Block:
         block = Block.get(block.client, _id=block.id)
     context.chat_history.file.refresh()
     return block
+
+
+def generate_action_choices(context: AgentContext) -> Block:
+
+    game_state = get_game_state(context)
+    quest_name = game_state.current_quest
+
+    prompt = (
+        f"Generate a multiple choice set of four options for the user to select {game_state.player.name}'s next "
+        f'action. The actions should be relevant to the story and include a single "custom" option that '
```

Comment:
question[non-blocking]: Given the required structure of the web just sending the text content back, I am curious why you would ask the LLM to generate the 'custom' option, rather than either injecting it manually into the json dict, or leaving it out entirely?

Reply:
I'm totally fine with dropping it (or manually injecting it). I thought it _might_ make the UI's job easier if it didn't have to think, but it is totally unnecessary from a token consumption perspective. I'm not even sure we want/need the map (a list should suffice). Thoughts?